aspnetcore/host-and-deploy/health-checks/samples/3.x/HealthChecksSample/CustomWriterStartup.cs

```diff
@@ -49,11 +54,53 @@ public void Configure(IApplicationBuilder app)
             });
         }
 
-        #region snippet_WriteResponse
-        private static Task WriteResponse(HttpContext httpContext, 
-            HealthReport result)
+#if SYSTEM_TEXT_JSON
+        #region snippet_WriteResponse_SystemTextJson
+        private static Task WriteResponse(HttpContext context, HealthReport result)
+        {
+            context.Response.ContentType = "application/json";
+
+            var options = new JsonWriterOptions
+            {
+                Indented = true
+            };
+
+            using (var stream = new MemoryStream())
+            {
+                using (var writer = new Utf8JsonWriter(stream, options))
+                {
+                    writer.WriteStartObject();
+                    writer.WriteString("status", result.Status.ToString());
+                    writer.WriteStartObject("results");
+                    foreach (var entry in result.Entries)
+                    {
+                        writer.WriteStartObject(entry.Key);
+                        writer.WriteString("status", entry.Value.Status.ToString());
+                        writer.WriteString("description", entry.Value.Description);
+                        writer.WriteStartObject("data");
+                        foreach (var item in entry.Value.Data)
+                        {
+                            var value = Convert.ToDecimal(item.Value);
+                            writer.WriteNumber(item.Key, value);
```

Comment:
This isn't right. The reason why you're having a problem, with this is that there's no guarantees about what kind of object `item.Value` will be.

The right incantation would be:

```C#
writer.WritePropertyName(item.Key);
JsonSerializer.Serialize(writer, item.Value, item.Value?.GetType() ?? typeof(object));
```

Reply:
I understand what ur saying, but I (perhaps incorrectly) assumed that this example wasn't meant to cover a delegate *in general*. I thought that **_this_** delegate for **_this_** health check wouldn't ever have anything in `entry.Value` that wouldn't be castable into a decimal (given the `HealthCheckResult` from the `MemoryHealthCheck`). Therefore, I wasn't concerned (*until now* lol). I understand that you provided code that won't break in general, and I understand that it's better to show it this way. Perhaps, it's never safe to assume anything about an object's type in a scenario like this. I'll keep that in mind going forward.

Reply:
>Perhaps, it's never safe to assume anything about an object's type in a scenario like this. I'll keep that in mind going forward.

Cool, I think you're on track now 👍